components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/MgwDefinitionUtils.java

```diff
@@ -117,4 +137,42 @@ private static boolean isResourceAvailable(String basePath, String path, String
         }
         return true;
     }
+
+    /**
+     * To validate the provided definition.yaml follows the yaml syntax
+     *
+     * @param filePath file path to definition.yaml
+     */
+    private static void validateYamlSyntax(String filePath) {
+        File file = new File(filePath);
+        //to check the existence of definitions.yaml file
+        if (!file.exists()) {
+            throw GatewayCmdUtils.createValidationException("'definition.yaml' file does not exists.", LOGGER);
+        }
+        try {
+            OBJECT_MAPPER_YAML.readTree(file);
+            //if the provided definitions.yaml file does not follow the yaml syntax
+        } catch (IOException e) {
+            throw GatewayCmdUtils.createValidationException("'definitions.yaml file cannot be parsed as yaml document.",
+                    LOGGER);
+        }
+    }
+
+    /**
+     * To find out the api information which is not used for code generation but included in the definitions.yaml
+     *
+     * @param outStream print stream if needed
+     */
+    public static void FindNotUsedAPIInformation(PrintStream outStream) {
+        rootDefinition.getApis().getApisMap().forEach((k, v) -> {
+            if (!v.getIsUsed()) {
+                String msg = "[Warning] API '" + v.getTitle() + "' version: '" + v.getVersion() + "' is not used but " +
+                        "added to the " + GatewayCliConstants.PROJECT_DEFINITION_FILE + ".";
+                LOGGER.info(msg);
+                if(outStream != null){
+                    outStream.println(msg);
```

Comment:
Do we need this message in the console output, shall we move this to a verbose kind of logs

Reply:
added.